collect_app/src/androidTest/java/org/odk/collect/android/storage/StorageMigrationBannerTest.java

```diff
@@ -0,0 +1,36 @@
+package org.odk.collect.android.storage;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.odk.collect.android.activities.MainMenuActivity;
+import org.odk.collect.android.support.StorageMigrationNotPerformedRule;
+import org.odk.collect.android.support.pages.MainMenuPage;
+
+public class StorageMigrationBannerTest {
+
+    @Rule
+    public ActivityTestRule<MainMenuActivity> main = new ActivityTestRule<>(MainMenuActivity.class);
+
+    @Rule
+    public StorageMigrationNotPerformedRule storageMigrationNotPerformedRule = new StorageMigrationNotPerformedRule();
```

Comment:
Really liking this use of `@Rule`! I think it'd be good to still have the `ResetStateRule` and standard permssion rules for these tests just to make sure they are starting fresh.

Reply:
Done, but when it comes to the new rule I removed it because it was called too late. I needed to override `beforeActivityLaunched` like here https://github.com/opendatakit/collect/pull/3623/commits/f6073fa3ecd523166fd21dbace8654398e2b6379#diff-031d2e263a9e3d4064f608116ef282e4R34